druid/src/widget/widget.rs

```diff
@@ -184,6 +184,16 @@ pub trait Widget<T> {
     /// [`RenderContext`]: trait.RenderContext.html
     fn paint(&mut self, ctx: &mut PaintCtx, data: &T, env: &Env);
 
+    /// Utility callback invoked after all painting is done and submitted, to perform
+    /// post-painting actions.
+    ///
+    /// This is typically for advanced use only. Most widgets have no use for this. This
+    /// is currently only used for wgpu integration, to allow wgpu-based rendering to
+    /// reliably occur after all druid native rendering finished, to avoid non-deterministic
+    /// out-of-order rendering (parent covering child) between those two unsynchronized
+    /// rendering pipelines.
+    fn post_render(&mut self);
```

Comment:
It would probably be easier to create a `Lifecycle::PostRender` variant instead of a new method.